test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ERCPrecompileSuite.java

```diff
@@ -1240,6 +1244,290 @@ private HapiApiSpec getErc721OwnerOfFromErc20TokenFails() {
 				);
 	}
 
+	private HapiApiSpec directCallsWorkForERC20() {
+
+		final AtomicReference<String> tokenNum = new AtomicReference<>();
+
+		final var tokenName = "TokenA";
+		final var tokenSymbol = "FDFGF";
+		final var tokenDecimals = 10;
+		final var tokenTotalSupply = 5;
+		final var tokenTransferAmount = 3;
+
+		final var symbolTxn = "symbolTxn";
+		final var nameTxn = "nameTxn";
+		final var decimalsTxn = "decimalsTxn";
+		final var totalSupplyTxn = "totalSupplyTxn";
+		final var balanceOfTxn = "balanceOfTxn";
+		final var transferTxn = "transferTxn";
+
+		return defaultHapiSpec("DirectCallsWorkForERC20")
+				.given(
+						newKeyNamed(MULTI_KEY),
+						cryptoCreate(ACCOUNT).balance(100 * ONE_HUNDRED_HBARS),
+						cryptoCreate(RECIPIENT).balance(100 * ONE_HUNDRED_HBARS),
+						cryptoCreate(TOKEN_TREASURY),
+						tokenCreate(FUNGIBLE_TOKEN)
+								.tokenType(TokenType.FUNGIBLE_COMMON)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(tokenTotalSupply)
+								.name(tokenName)
+								.symbol(tokenSymbol)
+								.decimals(tokenDecimals)
+								.treasury(TOKEN_TREASURY)
+								.adminKey(MULTI_KEY)
+								.supplyKey(MULTI_KEY)
+								.exposingCreatedIdTo(tokenNum::set),
+						tokenAssociate(ACCOUNT, FUNGIBLE_TOKEN),
+						tokenAssociate(RECIPIENT, FUNGIBLE_TOKEN),
+						cryptoTransfer(moving(tokenTransferAmount, FUNGIBLE_TOKEN).between(TOKEN_TREASURY, ACCOUNT))
+				).when(withOpContext(
+						(spec, ignore) -> {
+							var tokenAddress = asHexedSolidityAddress(asToken(tokenNum.get()));
+							allRunFor(spec,
+									contractCallWithFunctionAbi(
+											tokenAddress,
+											getABIFor(FunctionType.FUNCTION, "name", "ERC20ABI")
+									).via(nameTxn),
+									contractCallWithFunctionAbi(
+											tokenAddress,
+											getABIFor(FunctionType.FUNCTION, "symbol", "ERC20ABI")
+									).via(symbolTxn),
+									contractCallWithFunctionAbi(
+											tokenAddress,
+											getABIFor(FunctionType.FUNCTION, "decimals", "ERC20ABI")
+									).via(decimalsTxn),
+									contractCallWithFunctionAbi(
+											tokenAddress,
+											getABIFor(FunctionType.FUNCTION, "totalSupply", "ERC20ABI")
+									).via(totalSupplyTxn),
+									contractCallWithFunctionAbi(
+											tokenAddress,
+											getABIFor(FunctionType.FUNCTION, "balanceOf", "ERC20ABI"),
+											asHexedSolidityAddress(spec.registry().getAccountID(ACCOUNT))
+									).via(balanceOfTxn),
+									contractCallWithFunctionAbi(
+											tokenAddress,
+											getABIFor(FunctionType.FUNCTION, "transfer", "ERC20ABI"),
+											asHexedSolidityAddress(spec.registry().getAccountID(RECIPIENT)),
+											tokenTransferAmount
+									).via(transferTxn).payingWith(ACCOUNT)
+							);
```

Comment:
"ERC20ABI" can be extracted in a single variable. If you used ERC20ABI contract, haven't you missed uploading the .sol, .bin and .json files?